packages/react/src/components/ContentSwitcher/ContentSwitcher.js

```diff
@@ -5,125 +5,124 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import PropTypes from 'prop-types';
-import React from 'react';
-import classNames from 'classnames';
 import { settings } from 'carbon-components';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React, {
+  Children,
+  cloneElement,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { composeEventHandlers } from '../../tools/events';
 import { getNextIndex, matches, keys } from '../../internal/keyboard';
 
 const { prefix } = settings;
 
-export default class ContentSwitcher extends React.Component {
-  /**
-   * The DOM references of child `<Switch>`.
-   * @type {Array<Element>}
-   * @private
-   */
-  _switchRefs = [];
-
-  state = {};
-
-  static propTypes = {
-    /**
-     * Pass in Switch components to be rendered in the ContentSwitcher
-     */
-    children: PropTypes.node,
-
-    /**
-     * Specify an optional className to be added to the container node
-     */
-    className: PropTypes.string,
-
-    /**
-     * Specify an `onChange` handler that is called whenever the ContentSwitcher
-     * changes which item is selected
-     */
-    onChange: PropTypes.func.isRequired,
-
-    /**
-     * Specify a selected index for the initially selected content
-     */
-    selectedIndex: PropTypes.number,
-  };
-
-  static defaultProps = {
-    selectedIndex: 0,
-  };
-
-  static getDerivedStateFromProps({ selectedIndex }, state) {
-    const { prevSelectedIndex } = state;
-    return prevSelectedIndex === selectedIndex
-      ? null
-      : {
-          selectedIndex,
-          prevSelectedIndex: selectedIndex,
-        };
+function ContentSwitcher({
+  children,
```

Comment:
do we want to retain the  comments/docs?

Reply:
Which ones?

Reply:
these comments (it's different syntax so maybe inappropriate?)

![Screen Shot 2019-07-08 at 11 23 31 AM](https://user-images.githubusercontent.com/3846271/60826183-f0372400-a172-11e9-8832-c4c152e4ec3d.png)

it looks like you have them in the react version, but this diff view makes it look to me like these comments are not here in the vanilla one

Reply:
They should be at the bottom of the file on `ContentSwitcher.propTypes`!

Reply:
@joshblack oh yup sorry, quick morning read of code and i did not glance down far enough before making this comment 😑